components/script/dom/gpudevice.rs

```diff
@@ -73,6 +90,13 @@ pub struct GPUDevice {
     label: DomRefCell<Option<DOMString>>,
     device: webgpu::WebGPUDevice,
     default_queue: Dom<GPUQueue>,
+    error_scopes: DomRefCell<HashMap<u64, ErrorScopeInfo>>,
+    scope_stack: DomRefCell<Vec<u64>>,
```

Comment:
don't we mostly need to access all of the scope fields? Perhaps, we could put them in a struct, like `ScopeContext` and store the whole thing under `DomRefCell`